### ISSUE TITLE
Proposal: Add HowTo in docs for adding third party components to the application

### ISSUE THREAD

**Issue by markjritchie** [angular/angular-cli#387]
I'm from a .NET and AngularJS background who is now trying to embrace the Node / CLI development way of client side builds. There have been times I've struggled to do "easy" things such as adding the ng-table component to my angular-cli app. At this point in time I've still not actually got it working because I've been forced to resolve other issues first.

However I know I cannot be the only person with a similar background and struggling in a similar fashion. While I love what the CLI is offering I still get overwhelmed with all the new tech that I don't know as well as I should. I feel it would be prudent if we could support users of the CLI in adding components.

This could be achieved in its simplest form by adding a tutorial for how to do it in the docs, or in a fuller manner by adding something like ng install-component which could do the npm install and then update the build files so that its ready to be used.

If this type of functionality is already present I'm sorry that I missed it

**Comment by filipesilva** [comment]
No, you're definitely not the only person struggling with adding third party stuff. There's been quite a few questions of the sort (like https://github.com/angular/angular-cli/issues/330), and at the moment it's not very easy.

@hansl and @jkuri have been doing loads of work on this, they are likely the best people to elaborate on this point.

**Comment by bryanerayner** [comment]
It seems to me that the way this is set up, I should be able to add a 3rd party module (like `redux`) into `dist/vendor/`. Then, modify `system-config.js` and add the path. Am I right, or wrong on this?

Ideally, it would be nice to have a browserifyesque solution - Just install an npm module, and it goes to all the right places. Is that possible? What would the ideal manual steps be, for adding `redux` (for example) to an application?

**Comment by hansl** [comment]
To add 3rd party files you need to add them to `vendorNpmFiles`, then configure `system-config.ts` as you suggested. Adding `redux` is definitely something that should work.

Having the second solution (smartly extract files from node_modules) is part of our greater third party efforts, but we're blocked on TypeScript 2.0 for now.

**Comment by bryanerayner** [comment]
Sorry @hansl , still lost. What do you mean by `vendorNpmFiles`? Not seeing a file or directory matching that name.

Glad to see that there is some progress being made in this direction. In the meantime, it would be helpful to work down a few easy to follow steps that we can use in the mean-time, perhaps add it to the Readme?

**Comment by hansl** [comment]
It's in the `angular-cli-build.js`.

**Comment by jkuri** [comment]
@bryanerayner did you checked out the wiki?

**Comment by filoxo** [comment]
Is the [wiki page](https://github.com/angular/angular-cli/wiki/3rd-party-libs) missing a step? I followed all of them but then also needed to drop in a script ref to the `/vendor/lib/lib.js` in `index.html`. As far as I can tell, registering them into system-config.ts and vendorNpmFiles just tells systemjs where to find them to copy them over, but won't "bundle" them into the main app file. I would edit the page to reflect this but I just want to confirm that this is correct.

**Comment by immysl** [comment]
tried it according to the wiki but sadly doesn't generate a file in the vendor folder :(

**Comment by Infining** [comment]
Tried the example in the wiki and ran into the same problem. No file in vendor folder.

**Comment by bryanerayner** [comment]
The solution is to go to `angular-cli-build.js` and add your glob to the existing list. It will then be copied to `vendor/`

**Comment by filipesilva** [comment]
The [wiki page](https://github.com/angular/angular-cli/wiki/3rd-party-libs) example for momentjs was updated yesterday, and should be working correctly as of beta.5.
